### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/ssumpw`

### PR DESCRIPTION
**Pull request by Snehil-Shah**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds C `ndarray` API for `blas/ext/base/ssumpw`.
-   refactors package to follow updated project conventions.
-   Updates stride handling where stride is equal to zero.

## Related Issues

> Does this pull request have any related issues?

No.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/main.c [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			s2 += X[ ix+(2*strideX) ];
+			s3 += X[ ix+(3*strideX) ];
+			s4 += X[ ix+(4*strideX) ];
+			s5 += X[ ix+(5*strideX) ];
+			s6 += X[ ix+(6*strideX) ];
+			s7 += X[ ix+(7*strideX) ];
+			ix += 8 * strideX;
+		}
+		// Pairwise sum the accumulators:
+		sum = ((s0+s1) + (s2+s3)) + ((s4+s5) + (s6+s7));
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/main.c]:
```suggestion
		sum = ( (s0+s1) + (s2+s3) ) + ( (s4+s5) + (s6+s7) );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/addon.c [hunk 2]
```diff
@@ -17,10 +17,12 @@
...
 
 #include "stdlib/blas/ext/base/ssumpw.h"
+#include "stdlib/blas/base/shared.h"
 #include "stdlib/napi/export.h"
 #include "stdlib/napi/argv.h"
 #include "stdlib/napi/argv_int64.h"
 #include "stdlib/napi/argv_strided_float32array.h"
+#include "stdlib/napi/create_double.h"
 #include <node_api.h>
 #include <assert.h>
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/addon.c]:
```suggestion
```

**Review by headlessNode — CHANGES_REQUESTED** [review]
Aside from these two, looks good overall to me.

File: lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/main.c (hunk 1 above)
Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/main.c]: It feels more readable without the spaces, WDYT? I suppose that's why they weren't there in the [initial implementation](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/ssumpw/src/ssumpw.c#L101) as well.

Comment (headlessNode) [reply lib/node_modules/@stdlib/blas/ext/base/ssumpw/src/main.c]: I think it depends on personal taste. Some might find visual breaks more readable. But my suggestion was based on keeping things consistent across the packages in the `blas/ext/base`.

**Review by Planeshifter — APPROVED** [review]
PR looks great. Thanks @Snehil-Shah for your work and @headlessNode for the review!

**Pull request merged**